generators/spring-data-relational/support/database-data.ts

```diff
@@ -151,6 +153,7 @@ const databaseData: Record<string, DatabaseDataSpec> = {
     jdbcDriver: 'org.postgresql.Driver',
     hibernateDialect: 'org.hibernate.dialect.PostgreSQLDialect',
     port: ':5432/',
+    defaultPassword: 'postgresql',
```

Comment:
password is [required for databases](https://github.com/spring-projects/spring-boot/blob/ce01dbbad41496a2de6fb1c35bd08edfb8a8a430/spring-boot-project/spring-boot-docker-compose/src/test/java/org/springframework/boot/docker/compose/service/connection/mariadb/MariaDbEnvironmentTests.java#L59-L63).

Reply:
Actually is required for postgresql only.
Fixed in v3.4 branch https://github.com/spring-projects/spring-boot/pull/41511?notification_referrer_id=NT_kwDOABWf37MxMTUyMzQ1NDEzMDoxNDE3MTgz#issuecomment-2346590381